projects/oatpp/fuzz_mapper.cpp

```diff
@@ -0,0 +1,40 @@
+/* Copyright 2020 Google Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "oatpp/parser/json/mapping/ObjectMapper.hpp"
+#include "oatpp/core/macro/codegen.hpp"
+
+typedef oatpp::parser::Caret ParsingCaret;
+typedef oatpp::parser::json::mapping::Serializer Serializer;
+typedef oatpp::parser::json::mapping::Deserializer Deserializer;
+
+#include OATPP_CODEGEN_BEGIN(DTO)
+
+class EmptyDto : public oatpp::DTO {
+  DTO_INIT(EmptyDto, DTO)
+};
+
+
+class Test1 : public oatpp::DTO {
+  DTO_INIT(Test1, DTO)
+  DTO_FIELD(String, strF);
+};
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  std::string input(reinterpret_cast<const char*>(data), size);
```

Comment:
You can also do like this:

```cpp
oatpp::String input(reinterpret_cast<const char*>(data), size, true);
oatpp::parser::json::mapping::ObjectMapper mapper;
try {
  mapper.readFromString<oatpp::Object<Test1>>(input);
} catch(...) {}
```

In this case, only one string will be allocated

Reply:
Thanks! I fixed this up. Would you be happy to get the fuzzers moved upstream so they live in the oatpp repository?

Reply:
> Would you be happy to get the fuzzers moved upstream so they live in the oatpp repository?

Sounds interesting. Let's try to do this.